File: testdata/jennies/rawtypes/disjunctions/PHPRawTypes/src/Disjunctions/SomeStruct.php
<?php

namespace Grafana\Foundation\Disjunctions;

class SomeStruct implements \JsonSerializable
{
    public string $type;

    /**
     * @var mixed
     */
    public $fieldAny;

    /**
     * @param mixed|null $fieldAny
     */
    public function __construct( $fieldAny = null)
    {
        $this->type = "some-struct";
    
        $this->fieldAny = $fieldAny ?: null;
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->type !== $other->type) {
            return false;
        }
    
        if ($this->fieldAny !== $other->fieldAny) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{Type?: string, FieldAny?: mixed} $inputData */
        $data = $inputData;
        return new self(
            fieldAny: $data["FieldAny"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->Type = $this->type;
        $data->FieldAny = $this->fieldAny;
        return $data;
    }
}

File: testdata/jennies/rawtypes/disjunctions/JavaRawTypes/disjunctions/SomeStruct.java
package disjunctions;

import java.util.Objects;

public class SomeStruct {
    public String type;
    public Object fieldAny;
    public SomeStruct() {
        this.type = "";
        this.fieldAny = new Object();
    }
    public SomeStruct(String type,Object fieldAny) {
        this.type = type;
        this.fieldAny = fieldAny;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof SomeStruct)) return false;
        SomeStruct o = (SomeStruct) other;
        if (!Objects.equals(this.type, o.type)) return false;
        if (!Objects.equals(this.fieldAny, o.fieldAny)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.type, this.fieldAny);
    }
}
